common/src/main/java/com/microsoft/identity/common/internal/controllers/IAccountCallback.java

```diff
@@ -0,0 +1,39 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common.internal.controllers;
+
+public interface IAccountCallback<T> {
```

Comment:
I would consider creating a new interface which `extends TaskCompletedCallback` and adds the `onError` method -- then call it something like `TaskCompletedCallbackWithError` OBO.

I suggest having it accept a second type... something like...

```java
interface TaskCompletedCallbackWithError<T, U> extends TaskCompletedCallback<T> {
    void onError(U error);
}
```

Optionally, you could add a lower bound on the wildcard
```java
interface TaskCompletedCallbackWithError<T, ? extends Throwable> extends TaskCompletedCallback<T>
```

Reply:
I suggest all this because this `interface` is generic but its naming _is not_